src/error.rs

```diff
@@ -9,83 +9,61 @@ use std::str::FromStr;
 use url::ParseError;
 
 /// An error occurred while attempting to establish a session for a new `Client`.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum NewSessionError {
     /// The given WebDriver URL is invalid.
-    BadWebdriverUrl(ParseError),
+    #[error("webdriver url is invalid: {0}")]
```

Comment:
I think the general guidance from the error working group is that you _either_ print the inner error in the message _or_ you propagate it using `Error::source` (which both `#[from]` and `#[source]` do I believe). This currently does both. Let's remove `: {0}` here and stick with just exposing source.